daisi/src/cpps/logical/order_management/simple_order_management.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 The SOLA authors
+//
+// This file is part of DAISI.
+//
+// DAISI is free software: you can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation; version 2.
+//
+// DAISI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+// the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with DAISI. If not, see
+// <https://www.gnu.org/licenses/>.
+//
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "simple_order_management.h"
+
+using namespace daisi::material_flow;
+namespace daisi::cpps::logical {
+
+template <class> inline constexpr bool always_false_v = false;
+
+SimpleOrderManagement::SimpleOrderManagement(const AmrDescription &amr_description,
+                                             const Topology &topology,
+                                             const daisi::util::Pose &pose)
+    : OrderManagement(amr_description, topology, pose),
+      current_task_(),
+      current_ordering_(),
```

Comment:
Maybe `current_ordering_` could be called `queue_`, because this is basically what the SOM implements.